tools/js-sdk-release-tools/src/common/utils.ts

```diff
@@ -298,6 +299,31 @@ export async function resolveOptions(typeSpecDirectory: string): Promise<Exclude
     return options
 }
 
+export function updateApiVersionInTspConfig(typeSpecDirectory: string, apiVersion?: string) {
+    if (!apiVersion || apiVersion === '') {
+        logger.warn(`No apiVersion provided, skipping update to tspconfig.yaml.`);
+        return;
+    }
+    
+    const tspConfigPath = path.join(typeSpecDirectory, 'tspconfig.yaml');
+    if (!fs.existsSync(tspConfigPath)) {
+        throw new Error(`tspconfig.yaml not found at path: ${tspConfigPath}`);
+    }
+
+    const tspConfigContent = fs.readFileSync(tspConfigPath, 'utf8');
+    const tspConfig = yamlLoad(tspConfigContent);
+
+    if (!tspConfig || typeof tspConfig !== 'object') {
+        throw new Error('Invalid tspconfig.yaml format.');
+    }
+
+    tspConfig['options']['@azure-tools/typespec-ts']['api-version'] = apiVersion;
```

Comment:
should check if sub-object exists
e.g. check `tspConfig['options']['@azure-tools/typespec-ts']` exists